src/routes.rs

```diff
@@ -1922,3 +2057,338 @@ pub(crate) async fn unlock(
+                        proportional_millionths: config.fee_proportional_millionths,
+                    },
+                }])
+            })
+            .collect();
+
+        let first_leg = get_route(
+            &unlocked_state.channel_manager,
+            &unlocked_state.router,
+            unlocked_state.channel_manager.get_our_node_id(),
+            taker_pk,
+            if swaptype.is_buy() { Some(swaptype.amount_msats()) } else { None },
+            if swaptype.is_buy() { None } else { Some(swapstring.asset_id) },
+            vec![],
+        );
+        let second_leg = get_route(
+            &unlocked_state.channel_manager,
+            &unlocked_state.router,
+            taker_pk,
+            unlocked_state.channel_manager.get_our_node_id(),
+            if swaptype.is_buy() { Some(HTLC_MIN_MSAT) } else { Some(swaptype.amount_msats()) },
+            if swaptype.is_buy() { Some(swapstring.asset_id) } else { None },
+            receive_hints,
+        );
+
+        let (mut first_leg, mut second_leg) = match (first_leg, second_leg) {
+            (Some(f), Some(s)) => (f, s),
+            (Some(_), _) => {
+                return Err(APIError::NoRoute);
+            }
+            (_, Some(_)) => {
+                return Err(APIError::NoRoute);
+            }
+            _ => {
+                return Err(APIError::NoRoute);
+            }
+        };
+
+        // Set swap flag
+        second_leg.paths[0].hops[0].short_channel_id |= IS_SWAP_SCID;
+
+        // Generally in the last hop the fee_amount is set to the payment amount, so we need to
+        // override it depending on what type of swap we are doing
+        if let SwapType::BuyAsset { .. } = swaptype {
+            first_leg.paths[0].hops.last_mut().expect("Path not to be empty").fee_msat = HTLC_MIN_MSAT;
+        } else {
+            first_leg.paths[0].hops.last_mut().expect("Path not to be empty").fee_msat = 0;
+        }
+
+        let fullpaths = first_leg.paths[0]
+            .hops
+            .clone()
+            .into_iter()
+            .map(|mut hop| {
+                if let SwapType::SellAsset { amount_rgb, .. } = swaptype {
+                    hop.rgb_amount = Some(amount_rgb);
+                    hop.payment_amount = HTLC_MIN_MSAT;
+                }
+                hop
+            })
+            .chain(second_leg.paths[0].hops.clone().into_iter().map(|mut hop| {
+                if let SwapType::BuyAsset { amount_rgb, .. } = swaptype {
+                    hop.rgb_amount = Some(amount_rgb);
+                    hop.payment_amount = HTLC_MIN_MSAT;
+                }
+                hop
+            }))
+            .collect::<Vec<_>>();
+
+        let route = Route {
+            paths: vec![LnPath { hops: fullpaths, blinded_tail: None }],
+            route_params: Some(RouteParameters{
+                payment_params: PaymentParameters::for_keysend(unlocked_state.channel_manager.get_our_node_id(), 40, false),
```

Comment:
I believe this value is also not used because the cltv_delta is already part of the route hops we pre-computed. We could set it to 14 for consistency with the router, but in practice I think that's already the value being used.

Reply:
maybe setting it to 14 is a better choice (I propose to add a constant for this value and use that in both places)